src/ExpressionEvaluator/Core/Source/ResultProvider/Helpers/HashSetBuilder.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable disable
+
+#pragma warning disable CA1825 // Avoid zero-length array allocations.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis
+{
+    // Based on the ArrayBuilder.cs implementation.
+    internal sealed class HashSetBuilder<T, TComparer>
+        where TComparer : IEqualityComparer<T>, new()
+    {
+        private static readonly ObjectPool<HashSetBuilder<T, TComparer>> s_poolInstance =
+            new ObjectPool<HashSetBuilder<T, TComparer>>(() => new HashSetBuilder<T, TComparer>(new TComparer()), 16);
+
+        // Note: the value of this dictionary is unused. This is just being used as a HashSet
+        // (CLR 2.0 doesn't have an actual HashSet).
```

Comment:
ResultProvider.csproj is now built with netstandard2.0 so HashSet<T> should be available.